RInAppMessaging/Classes/Views/Presenters/BaseViewPresenter.swift

```diff
@@ -20,16 +19,7 @@ internal class BaseViewPresenter: BaseViewPresenterType {
 
     var campaign: Campaign!
     var impressions: [Impression] = []
-    lazy var associatedImage: UIImage? = {
-        guard let imageURLString = campaign.data.messagePayload.resource.imageUrl,
-            let encodedImageURLString = imageURLString.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed),
-            let encodedImageURL = URL(string: encodedImageURLString),
-            let imageData = try? Data(contentsOf: encodedImageURL) else {
-
-            return nil
-        }
-        return UIImage(data: imageData)
-    }()
+    var associatedImage: UIImage?
```

Comment:
I liked that we fetched this lazily before.

I think now we risk delaying the campaign display to a point where it could be out of context of what the app is now doing.

Reply:
This change though is similar to how [Android downloads then caches the image](https://github.com/rakutentech/android-inappmessaging/blob/73df94dac6f81783c09488a10ff00871432def4b/inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/service/DisplayMessageJobIntentService.kt#L61) before opening up the view. We could even argue this new way is even **faster (i.e. less delay) than before**, at least for the `maxImpressions>1` case, due to the cache being used here

Reply:
@donnie-jp previously it was also delayed. `associatedImage` was called before displaying and `UIImage(data: imageData)` is a synchronous method. Now it's just another async call on the same queue

Reply:
Ah I forgot that the point was to avoid displaying the campaign if the image can't be loaded 😐